### PR TITLE
Circle Constructor (Circle, ThickCircle, SolidCircle)

### PR DESCRIPTION
**Pull request by kammitama5**
Something along these lines...

### CODE REVIEW

**Comment by kammitama5** [comment]
So ```pathDrawer``` needs Points. This is going to be like ```rectangle```, isn't it?

File: codeworld-api/src/CodeWorld/Driver.hs [hunk 1]
```diff
@@ -219,6 +219,9 @@ pictureToDrawing (SolidRectangle _ w h) = Shape $ polygonDrawer (rectangleVertic
 pictureToDrawing (ThickRectangle _ lw w h) = Shape $ pathDrawer (rectangleVertices w h) lw True False
 pictureToDrawing (ClosedCurve _ pts) = Shape $ pathDrawer pts 0 True True
 pictureToDrawing (ThickClosedCurve _ pts w) = Shape $ pathDrawer pts w True True
+pictureToDrawing (Circle _ r) = Shape $ pathDrawer r 0 True False
```
[the remark this answers is not in the document]
Comment (kammitama5) [reply codeworld-api/src/CodeWorld/Driver.hs]: Yes. It does now. Thanks!

File: codeworld-api/src/CodeWorld/Picture.hs [hunk 2]
```diff
@@ -225,11 +232,13 @@ thickRectangle lw w h = ThickRectangle callStack lw w h
 
 -- | A thin circle, with this radius
 circle :: HasCallStack => Double -> Picture
-circle = arc 0 (2 * pi)
+circle r = Circle callStack r
```
Comment (Gabriel439) [inline codeworld-api/src/CodeWorld/Picture.hs]: You can simplify this to:

```haskell
circle = Circle callstack
```

File: codeworld-api/src/CodeWorld/Picture.hs [hunk 3]
```diff
@@ -225,11 +232,13 @@ thickRectangle lw w h = ThickRectangle callStack lw w h
...
 -- | A thin circle, with this radius
 circle :: HasCallStack => Double -> Picture
-circle = arc 0 (2 * pi)
+circle r = Circle callStack r 
+-- circle = arc 0 (2 * pi)
 
 -- | A thick circle, with this line width and radius
 thickCircle :: HasCallStack => Double -> Double -> Picture
-thickCircle w = thickArc w 0 (2 * pi)
+thickCircle r lw = ThickCircle callStack r lw
```
Comment (Gabriel439) [inline codeworld-api/src/CodeWorld/Picture.hs]: `thickCircle = ThickCircle callStack`

File: codeworld-api/src/CodeWorld/Picture.hs [hunk 4]
```diff
@@ -246,7 +255,9 @@ thickArc w b e r = ThickArc callStack b e r w
 
 -- | A solid circle, with this radius
 solidCircle :: HasCallStack => Double -> Picture
-solidCircle = sector 0 (2 * pi)
+solidCircle r = SolidCircle callStack r
```
Comment (Gabriel439) [inline codeworld-api/src/CodeWorld/Picture.hs]: `solidCircle = SolidCircle callStack`

File: codeworld-api/src/CodeWorld/Picture.hs [hunk 5]
```diff
@@ -246,7 +255,9 @@ thickArc w b e r = ThickArc callStack b e r w
...
+
+--solidCircle = sector 0 (2 * pi)
```
Comment (Gabriel439) [inline codeworld-api/src/CodeWorld/Picture.hs]: Also, you don't need these comments any longer

File: codeworld-api/src/CodeWorld/Picture.hs (hunk 2 above)
Comment (kammitama5) [reply codeworld-api/src/CodeWorld/Picture.hs]: is this the eta reduction? :D

Comment (Gabriel439) [reply codeworld-api/src/CodeWorld/Picture.hs]: Yes :)

The general trick is that if you define some function like:

```haskell
f x = g x
```

... you can simplify it to:

```haskell
f = g
```

For example, if you define:

```haskell
incrementEveryNumber numbers = map (+ 1) numbers
```

... you can simplify that to:

```haskell
incrementEveryNumber = map (+ 1)
```

File: codeworld-api/src/CodeWorld/Driver.hs [hunk 6]
```diff
@@ -219,6 +219,9 @@ pictureToDrawing (SolidRectangle _ w h) = Shape $ polygonDrawer (rectangleVertic
 pictureToDrawing (ThickRectangle _ lw w h) = Shape $ pathDrawer (rectangleVertices w h) lw True False
 pictureToDrawing (ClosedCurve _ pts) = Shape $ pathDrawer pts 0 True True
 pictureToDrawing (ThickClosedCurve _ pts w) = Shape $ pathDrawer pts w True True
+pictureToDrawing (Circle _ r) = Shape $ arcDrawer 0 (2 * pi) r 0
+pictureToDrawing (SolidCircle _ r) = Shape $ sectorDrawer 0 (2 * pi) r 
+pictureToDrawing (ThickCircle _ r lw) = Shape $ arcDrawer lw (2 * pi) r 0
```
Comment (cdsmith) [inline codeworld-api/src/CodeWorld/Driver.hs]: Double check the order of parameters here?  `arcDrawer` has parameters `b`, `e`, `r`, `w`, which should mean "begin angle", "end angle", "radius", and "line width", in that order.

**Comment by kammitama5** [comment]
Is there anything you'd like me to fix between 544-568 of ```Driver.hs```? I know they're not really used, but...

**Comment by cdsmith** [comment]
Yeah, we can clean up the properties later, as we talked about in the last meeting.

**Pull request merged**